File: Tugas Akhir Semester 1/TrashBinFe/app/src/main/java/com/trashbin/app/data/model/Order.kt
package com.trashbin.app.data.model

import com.google.gson.annotations.SerializedName

data class Order(
    @SerializedName("id") val id: Int,
    @SerializedName("buyer_id") val buyerId: Int,
    @SerializedName("seller_id") val sellerId: Int,
    @SerializedName("listing_id") val listingId: Int,
    @SerializedName("quantity") val quantity: Int,
    @SerializedName("total_price") val totalPrice: Double,
    @SerializedName("shipping_address") val shippingAddress: String,
    @SerializedName("notes") val notes: String?,
    @SerializedName("status") val status: String, // pending, confirmed, shipped, completed, cancelled
    @SerializedName("buyer_rating") val buyerRating: Int?,
    @SerializedName("buyer_review") val buyerReview: String?,
    @SerializedName("seller_rating") val sellerRating: Int?,
    @SerializedName("seller_review") val sellerReview: String?,
    @SerializedName("buyer") val buyer: User,
    @SerializedName("seller") val seller: User,
    @SerializedName("listing") val listing: MarketplaceListing,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

File: Tugas Akhir Semester 1/TrashBinBe/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Order extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'listing_id',
        'buyer_id',
        'seller_id',
        'quantity',
        'total_price',
        'status',
        'payment_status',
        'shipping_address',
        'notes',
        'rating',
        'review',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'quantity' => 'decimal:2',
        'total_price' => 'decimal:2',
        'rating' => 'integer',
    ];

    /**
     * Get the listing for this order
     */
    public function listing()
    {
        return $this->belongsTo(MarketplaceListing::class, 'listing_id');
    }

    /**
     * Get the buyer of this order
     */
    public function buyer()
    {
        return $this->belongsTo(User::class, 'buyer_id');
    }

    /**
     * Get the seller of this order
     */
    public function seller()
    {
        return $this->belongsTo(User::class, 'seller_id');
    }
    
    /**
     * Scope to get pending orders
     */
    public function scopePending($query)
    {
        return $query->where('status', 'pending');
    }
    
    /**
     * Scope to get confirmed orders
     */
    public function scopeConfirmed($query)
    {
        return $query->where('status', 'confirmed');
    }
    
    /**
     * Scope to get completed orders
     */
    public function scopeCompleted($query)
    {
        return $query->where('status', 'completed');
    }
    
    /**
     * Scope to get cancelled orders
     */
    public function scopeCancelled($query)
    {
        return $query->where('status', 'cancelled');
    }
}
